ndk/utils.tsx

```diff
@@ -240,3 +252,192 @@ export const fetchEvents = async (
     relaySetSubscription.start();
   });
 };
+
+export const getNormalizedName = (pubkey: string, user?: NDKUser) => {
+  const profile = user?.profile;
+
+  return (
+    profile?.displayName ?? profile?.name ?? `${pubkey.substring(0, 5)}...`
+  );
+};
+
+export const getLnurlServiceEndpoint = (
+  zappedUserProfile?: NDKUserProfile,
+  zappedEvent?: NDKEvent
+) => {
+  let lud06: string | undefined;
+  let lud16: string | undefined;
+
+  if (zappedEvent) {
+    const zapTag = zappedEvent.getMatchingTags("zap")[0];
+
+    if (zapTag) {
+      switch (zapTag[2]) {
+        case "lud06":
+          lud06 = zapTag[1];
+          break;
+        case "lud16":
+          lud16 = zapTag[1];
+          break;
+        default:
+          throw new Error(`Unknown zap tag ${zapTag}`);
+      }
+    }
+  }
+
+  if (!lud06 && !lud16) {
+    lud06 = zappedUserProfile?.lud06;
+    lud16 = zappedUserProfile?.lud16;
+  }
+
+  if (lud16) {
+    const [name, domain] = lud16.split("@");
+    return `https://${domain}/.well-known/lnurlp/${name}`;
+  } else if (lud06) {
+    const { words } = bech32.decode(lud06, 1000);
+    const data = bech32.fromWords(words);
+    const utf8Decoder = new TextDecoder("utf-8");
+    return utf8Decoder.decode(data);
+  }
+};
+
+const getZapEndpoint = async (
+  zappedUserProfile?: NDKUserProfile,
+  zappedEvent?: NDKEvent
+) => {
+  const lnurlServiceEndpoint = getLnurlServiceEndpoint(
+    zappedUserProfile,
+    zappedEvent
+  );
+
+  if (!lnurlServiceEndpoint) {
+    return;
+  }
+
+  const { data } = await axios(lnurlServiceEndpoint);
+
+  if (data?.allowsNostr && (data?.nostrPubkey || data?.nostrPubKey)) {
+    return data.callback;
+  }
+};
+
+const getUserRelayUrls = async (
+  user: NDKUser,
+  opts?: { filter?: "readable" | "writable" }
+) => {
+  const relayListEvent = Array.from(await user.relayList())[0];
+  const { filter } = opts ?? {};
+
+  if (!relayListEvent) {
+    return null;
+  }
+
+  return relayListEvent
+    .getMatchingTags("r")
+    .filter((tag) => {
+      if (filter === "readable") {
+        return tag[2] === "read" || tag[2] === undefined;
+      }
+
+      if (filter === "writable") {
+        return tag[2] === "write" || tag[2] === undefined;
+      }
+
+      return true;
+    })
+    .map((tag) => tag[1]);
+};
+
+interface SignEventParams {
+  event: EventTemplate;
+  ndk: NDK;
+  isAnonymous?: boolean;
+}
+
+const signEvent = async ({ event, ndk, isAnonymous }: SignEventParams) => {
+  if (isAnonymous) {
+    return finishEvent(event, generatePrivateKey());
+  }
+
+  const zapRequestEvent = new NDKEvent(ndk, event as NostrEvent);
+
+  await zapRequestEvent.sign();
+
+  return await zapRequestEvent.toNostrEvent();
+};
```

Comment:
Rename zapRequestEvent to just event

Reply:
That's the name of one of the params though.

Reply:
I agree the name here doesn't make sense though 😂. Let me think about it.

Reply:
How about this?

```ts
const signEvent = async ({ eventTemplate, ndk, isAnonymous }: SignEventParams) => {
  if (isAnonymous) {
    return finishEvent(eventTemplate, generatePrivateKey());
  }

  const event = new NDKEvent(ndk, eventTemplate as NostrEvent);

  await event.sign();

  return await event.toNostrEvent();
};
```